xbmc/rendering/dx/DeviceResources.cpp

```diff
@@ -293,28 +300,6 @@ void DX::DeviceResources::CreateDeviceResources()
       &context                   // Returns the device immediate context.
     );
 
-  if (FAILED(hr))
```

Comment:
why this was removed?

Reply:
Because if you pass adapter parameter to the function the only valid conf will be with D3D_DRIVER_TYPE_UNKNOWN. Read https://msdn.microsoft.com/en-us/library/windows/desktop/ff476082(v=vs.85).aspx

Reply:
I know how device is creating. my question is why you remove creating WARP device in case if previous creation fails?

Reply:
if you pass D3D_DRIVER_TYPE_UNKNOWN the function never fail, if no hardware device, autofall to software rendering, but i think this case is very rare. But it is non directly relate to this issue.

Reply:
a driver may fails in any time just because a driver was written by an idiot.